### PR TITLE
SEPA Direct Debit Single Result Object Pattern

### PR DESCRIPTION
**Pull request by sarahkoop**
### Summary of changes

 - Update SEPA Direct Debit public methods with single result object callback pattern

 ### Checklist

 - [x] Added a changelog entry

### Authors

- @sarahkoop

### CODE REVIEW

File: SEPADirectDebit/src/main/java/com/braintreepayments/api/SEPADirectDebitPaymentAuthRequestParams.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package com.braintreepayments.api;
+
+import androidx.fragment.app.FragmentActivity;
```
Comment (saperi22) [inline SEPADirectDebit/src/main/java/com/braintreepayments/api/SEPADirectDebitPaymentAuthRequestParams.java:3]: unnecessary import?

File: SEPADirectDebit/src/main/java/com/braintreepayments/api/SEPADirectDebitPaymentAuthRequestParams.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+/**
+ * Returned via the {@link SEPADirectDebitPaymentAuthRequestCallback} after calling
+ * {@link SEPADirectDebitClient#createPaymentAuthRequest(SEPADirectDebitRequest, SEPADirectDebitPaymentAuthRequestCallback)}.
+ *
+ * Inspect the {@link SEPADirectDebitNonce} property to determine if tokenization is complete, or
+ * if you must continue the SEPA mandate web flow via
+ * {@link SEPADirectDebitLauncher#launch(FragmentActivity, SEPADirectDebitPaymentAuthRequest.ReadyToLaunch)}
+ */
+public class SEPADirectDebitPaymentAuthRequestParams {
```
Comment (saperi22) [inline SEPADirectDebit/src/main/java/com/braintreepayments/api/SEPADirectDebitPaymentAuthRequestParams.java:13]: QQ: I see that most of the newly created files are in kotlin. Any reason this is an exception?

File: SEPADirectDebit/src/main/java/com/braintreepayments/api/SEPADirectDebitPaymentAuthRequestParams.java (hunk 1 above)
Comment (saperi22) [reply SEPADirectDebit/src/main/java/com/braintreepayments/api/SEPADirectDebitPaymentAuthRequestParams.java:3]: I see that it is used in the comment.

File: SEPADirectDebit/src/main/java/com/braintreepayments/api/SEPADirectDebitPaymentAuthRequestParams.java (hunk 2 above)
Comment (sarahkoop) [reply SEPADirectDebit/src/main/java/com/braintreepayments/api/SEPADirectDebitPaymentAuthRequestParams.java:13]: Ah this file isn't actually new it was renamed from the previous SEPADirectDebitPaymentAuthRequest class (which is replaced by the new Kotlin sealed class in this PR). We are slowly migrating to Kotlin, so have been trying to add new files in Kotlin and convert small interfaces, but I didn't refactor this one into Kotlin yet.

File: SEPADirectDebit/src/main/java/com/braintreepayments/api/SEPADirectDebitInternalTokenizeCallback.java [hunk 3]
```diff
@@ -4,9 +4,9 @@
 
 /**
  * Callback for receiving result of
- * {@link SEPADirectDebitClient#tokenize(SEPADirectDebitPaymentAuthResult, SEPADirectDebitTokenizeCallback)}.
+ * {@link SEPADirectDebitClient#tokenize(SEPADirectDebitPaymentAuthResult, SEPADirectDebitInternalTokenizeCallback)}.
  */
-public interface SEPADirectDebitTokenizeCallback {
+public interface SEPADirectDebitInternalTokenizeCallback {
```
Comment (saperi22) [inline SEPADirectDebit/src/main/java/com/braintreepayments/api/SEPADirectDebitInternalTokenizeCallback.java:9]: What's the significance of the word `internal`?
We still expose this callback to the merchant, don't we?

File: SEPADirectDebit/src/test/java/com/braintreepayments/api/SEPADirectDebitClientUnitTest.java [hunk 4]
```diff
@@ -63,19 +63,21 @@ public void tokenize_onCreateMandateRequestSuccess_callsBackSEPAResponse_andSend
...
         verify(braintreeClient).sendAnalyticsEvent("sepa-direct-debit.selected.started");
 
         ArgumentCaptor<SEPADirectDebitPaymentAuthRequest> captor =
                 ArgumentCaptor.forClass(SEPADirectDebitPaymentAuthRequest.class);
-        verify(sepaFlowStartedCallback).onResult(captor.capture(), isNull());
+        verify(paymentAuthRequestCallback).onResult(captor.capture());
 
-        SEPADirectDebitPaymentAuthRequest sepaResponseResult = captor.getValue();
+        SEPADirectDebitPaymentAuthRequest paymentAuthRequest = captor.getValue();
+        assertTrue(paymentAuthRequest instanceof  SEPADirectDebitPaymentAuthRequest.ReadyToLaunch);
```
Comment (saperi22) [inline SEPADirectDebit/src/test/java/com/braintreepayments/api/SEPADirectDebitClientUnitTest.java:74]:
```suggestion
        assertTrue(paymentAuthRequest instanceof SEPADirectDebitPaymentAuthRequest.ReadyToLaunch);
```

**Review by saperi22 — APPROVED** [review]

File: SEPADirectDebit/src/main/java/com/braintreepayments/api/SEPADirectDebitInternalTokenizeCallback.java (hunk 3 above)
Comment (sarahkoop) [reply SEPADirectDebit/src/main/java/com/braintreepayments/api/SEPADirectDebitInternalTokenizeCallback.java]: Ah whoops that should have been made package-private - updated!

**Review by tdchow — APPROVED** [review]
:shipit:

File: SEPADirectDebit/src/test/java/com/braintreepayments/api/SEPADirectDebitClientUnitTest.java (hunk 4 above)
Comment (saperi22) [reply SEPADirectDebit/src/test/java/com/braintreepayments/api/SEPADirectDebitClientUnitTest.java:74]: ultra nit: looks like there's two spaces in there.

**Review by MaxHastingsPP — APPROVED** [review]

**Pull request merged**